ipld/src/selector/mod.rs

```diff
@@ -258,3 +374,49 @@ impl Selector {
         }
     }
 }
+
+fn replace_recursive_edge(
+    next_sel: Selector,
+    replace: Option<Box<Selector>>,
+) -> Option<Box<Selector>> {
```

Comment:
This function does not need to know about the `Box`es and actually becomes simpler if you leave them out here

Reply:
ah yeah, that was just an artifact of me refactoring to use Rc to avoid all the heap clones then reverting to save for after the full functionality. 

In a practical sense it's functionally is ~ the same, but yeah probably cleaner